src/simdutf/haswell/intrinsics.h

```diff
@@ -57,6 +57,10 @@ SIMDUTF_POP_DISABLE_WARNINGS
     // we roll our own
     #define _blsr_u64(n) ((n - 1) & n)
   #endif //  _blsr_u64
+  #ifndef _blsmsk_u32
+    // we roll our own
+    #define _blsmsk_u32(n) ((n - 1) ^ n)
```

Comment:
shouldn't this be `((n)-1)^(n)) `?

and the same protection for _blsr_u64 above.

or, avoiding preprocessor headache: `[](auto n){return (n - 1) & n;}();`

Reply:
Sure.

Reply:
I made the macros safer...  I am not very concerned about it because we use those only very rarely. And it is an internal hack.